lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -133,29 +131,29 @@ def check_assoc(assoc)
           end
         end
 
-        def add_resource_relationships(attrs, serializer, options = {})
-          options[:add_included] = options.fetch(:add_included, true)
-
-          serializer.associations.each do |association|
-            key = association.key
-            serializer = association.serializer
-            opts = association.options
-
-            attrs[:relationships] ||= {}
-
-            if serializer.respond_to?(:each)
-              add_relationships(attrs, key, serializer)
+        def resource_relationship_value(serializer, options = {})
+          if serializer.respond_to?(:each)
+            serializer.map { |s| resource_identifier(s) }
+          else
+            if options[:virtual_value]
```

Comment:
Maybe we could do something similar to JsonAdapter, that if found a instance of an model object it will query that object and use it as the relationship. I know that it might be an edge case, but would be nice to cover, taking into account the level o abstraction we are working with when trying to cover everyone using AMS :grin: